src/main.rs

```diff
@@ -41,18 +40,37 @@ fn main() {
                      {percent_change_7d}, {last_updated}",
                 ),
         )
+        .arg(
+            Arg::with_name("divider")
+                .short("d")
+                .long("divider")
+                .help("Enter the divider")
+                .takes_value(true)
+                .default_value(" | "),
+        )
         .get_matches();
 
-    let mut url = PathBuf::from("http://api.coinmarketcap.com/v1/ticker/");
-    url.push(matches.value_of("crypto").unwrap());
-
     let client = Client::new();
-    let mut response = client.get(url.to_str().unwrap()).send().unwrap();
+    let mut response = client
+        .get("http://api.coinmarketcap.com/v1/ticker/?limit=10")
+        .send()
+        .unwrap();
     let mut contents = String::new();
     response.read_to_string(&mut contents).unwrap();
-
     let stocks: Vec<HashMap<String, String>> = serde_json::from_str(&contents).unwrap();
-    let stock = stocks.first().unwrap();
 
-    println!("{}", strfmt(matches.value_of("format").unwrap(), stock).unwrap());
+    let cryptos: Vec<_> = matches.values_of("crypto").unwrap().collect();
+    let stocks: Vec<_> = stocks
+        .iter()
+        .filter(|&s| cryptos.contains(&s.get("symbol").unwrap().as_str()))
+        .collect();
+
+    let mut iter = stocks.iter().peekable();
+    while let Some(current) = iter.next() {
+        print!("{}", strfmt(matches.value_of("format").unwrap(), current).unwrap());
+        match iter.peek() {
+            Some(_) => print!("{}", matches.value_of("divider").unwrap()),
```

Comment:
It might be better to cache the `divider` and `format` values, I'm not sure if the compiler will be able to do that with those `.unwrap()`s.